### PR TITLE
Integrate RN Build 4/4/22

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Integrate RN Nightly Build 4/4/22 https://github.com/facebook/react-native/compare/8a5460ce8...d5da70e17

### Type of Change
- Integration

### What
- Adds W3CPointerExample to RNTester https://github.com/facebook/react-native/commit/50070ec7c64b4d9ad087606c2a4eee5eb80b9ca9
- Adds API to ReactRootView if JS touch events are needed https://github.com/facebook/react-native/commit/0a517ae43892fb764d829f8bae56c1ac58356b1b
- Upgrade Metro dependencies to 0.70.0 https://github.com/facebook/react-native/commit/8b81dea74e48ba0b0e0687afe4dd6528d5031d29
   - Verified Fast Refresh still works
- Optimized Virtualized List https://github.com/facebook/react-native/commit/19cf70266eb8ca151aa0cc46ac4c09cb987b2ceb
- Hermes Compiler packaged alongside RN releases https://github.com/facebook/react-native/commit/3c2ce2906fde32b6b3256d459e9795774f361b31
- Switches from INSTANTIATE_TEST_CASE_P to INSTANTIATE_TEST_SUITE_P in googletests https://github.com/facebook/react-native/commit/12a32549f427c993637d41569c49baf5ab7113e8
   - Had to override and file an issue https://github.com/microsoft/react-native-windows/issues/9791

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9789)

### CODE REVIEW

File: packages/@react-native-windows/virtualized-list/src/VirtualizedList.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ import type {
   ViewToken,
   ViewabilityConfigCallbackPair,
 } from './ViewabilityHelper';
-import type {LayoutEvent} from '../Types/CoreEventTypes';
+import type {LayoutEvent} from 'react-native/Libraries/Types/CoreEventTypes';
```
Comment (acoates-ms) [inline packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:37]: I think we use relative paths within the package.  - We should keep it relative, if nothing else they are less intensive on the bundler to work out the correct location.

Comment (TatianaKapos) [reply packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:37]: This was to fix a linting error, ` Cannot resolve module' ../Types/CoreEventTypes'. [cannot-resolve-module]`, I'm not sure of a way to make it relative while still passing the linter.

But we also import ScrollResponderType, ViewStyleProp and others in the same way. https://github.com/microsoft/react-native-windows/blob/827fa978fe5b706ca23b9beca5b9bcbd36dabb5d/packages/%40react-native-windows/virtualized-list/src/VirtualizedList.js#L30-L31

Comment (acoates-ms) [reply packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:37]: Those examples are from another package, so they have to use a module level import.

Comment (acoates-ms) [reply packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:37]: Maybe you need to modify .flowconfig?

**Review by chiaramooney — APPROVED** [review]
👍

**Pull request merged**